### PR TITLE
Adding amplification channel

### PR DESCRIPTION
**Pull request by kaspernielsen96**
**Context:**
Adding amplification channel
**Description of the Change:**
Adding the amplification channel and tests of the channel
**Benefits:**
More (louder) channels
**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: tests/test_lab_dev/test_transformations.py [hunk 1]
```diff
@@ -477,6 +478,50 @@ def test_representation(self):
         assert math.allclose(rep2.c, [1.0 + 0.0j])
 
 
+class TestAmplifier:
```
Comment (ziofil) [inline tests/test_lab_dev/test_transformations.py:481]: Let's add a test or two to verify correct operation. You can use identities for e.g. how an amplifier maps a coherent state, or how the Amplifier composes with the Attenuator.

Comment (kaspernielsen96) [reply tests/test_lab_dev/test_transformations.py:481]: I added a test for composing the amplifier and attenuator

File: tests/test_lab_dev/test_transformations.py [hunk 2]
```diff
@@ -477,6 +478,69 @@ def test_representation(self):
...
+    def test_operation(self):
+        amp_channel = Amplifier(modes=[0], gain=1.5)
+        att_channel = Attenuator(modes=[0], transmissivity=0.7)
+        operation = amp_channel >> att_channel
+
+        assert math.allclose(
+            operation.representation.A,
+            [
+                [
+                    [0.0 + 0.0j, 0.75903339 + 0.0j, 0.25925926 + 0.0j, 0.0 + 0.0j],
```
Comment (ziofil) [inline tests/test_lab_dev/test_transformations.py:533]: how was this matrix calculated?

Comment (kaspernielsen96) [reply tests/test_lab_dev/test_transformations.py:533]: It has been verified manually, but issues regarding .dual in MM has been found.

File: mrmustard/lab_dev/transformations/transformations.py [hunk 3]
```diff
@@ -389,6 +389,77 @@ def representation(self) -> Bargmann:
...
+    all the modes share the same gain.
+
+    .. code-block ::
+
+        >>> import numpy as np
+        >>> from mrmustard.lab_dev import Amplifier
+
+        >>> channel = Amplifier(modes=[1, 2], gain=1.1)
+        >>> assert channel.modes == [1, 2]
+        >>> assert np.allclose(channel.gain.value, [1.1, 1.1])
```
Comment (ziofil) [inline mrmustard/lab_dev/transformations/transformations.py:405]: please update these with an actual test of some kind
```suggestion
        >>> amp = Amplifier([0], gain=2)
        >>> coh = Coherent([0], x=1.0, y=2.0)  # units of hbar=2 (default) 
        >>> _, mu, _ = (coh >> amp).phase_space(0)
        >>> assert np.allclose(mu[0], np.array([2.0, 4.0]))
```

File: mrmustard/lab_dev/transformations/transformations.py [hunk 4]
```diff
@@ -389,6 +389,78 @@ def representation(self) -> Bargmann:
...
+                gain_bounds,
+                None,
+            )
+        )
+
+    @property
+    def representation(self) -> Bargmann:
+        n_modes = len(self.modes)
+        g = list(reshape_params(n_modes, g=self.gain.value))[0]
+        return Bargmann(*triples.amplifier_Abc(g))
```
Comment (ziofil) [inline mrmustard/lab_dev/transformations/transformations.py:461]:
```suggestion
        g = list(reshape_params(len(self.modes), g=self.gain.value))[0]
        return Bargmann(*triples.amplifier_Abc(g))
```

**Review by ziofil — APPROVED** [review]
Great! I added a couple tests. Feel free to merge when you want.

**Pull request merged**